src/chain/mod.rs

```diff
@@ -29,8 +32,15 @@ pub async fn export<D: Digest>(
 
     // Stream stateroots in range stateroot_lookup_limit..=tipset.epoch(). Also
     // stream all block headers until genesis.
-    let blocks =
-        stream_chain(&db, tipset.clone().chain(&db), stateroot_lookup_limit).with_seen(seen);
+    let blocks = par_buffer(
+        1024,
```

Comment:
Let's comment on the magic number. Did you try other values?

Reply:
I tried with 2k and saw no improvement. And 1024 is low enough not to waste too much memory.